### PR TITLE
[CLOUDP-229226] Search node management - implement list command

### PR DESCRIPTION
**Pull request by jeroenvervaeke**
## Proposed changes

Added `atlas cluster search nodes list --clusterName clusterName [--projectId projectId]` command.

_Jira ticket:_ CLOUDP-229226

### CODE REVIEW

File: internal/cli/atlas/search/nodes/list_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+func TestList_Run(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockStore := mocks.NewMockSearchNodesLister(ctrl)
+
+	listOpts := &ListOpts{
+		store: mockStore,
+	}
+
+	expected := &atlasv2.ApiSearchDeploymentResponse{}
```
Comment (blva) [inline internal/cli/atlas/search/nodes/list_test.go:34]: [nit] can we populate the response?

File: internal/cli/atlas/search/nodes/list.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+func (opts *ListOpts) Run() error {
+	r, err := opts.store.SearchNodes(opts.ConfigProjectID(), opts.clusterName)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+// mongocli atlas clusters search nodes list [--projectId projectId] [--clusterName name].
```
Comment (blva) [inline internal/cli/atlas/search/nodes/list.go:59]:
```suggestion
// atlas clusters search nodes list [--projectId projectId] [--clusterName name].
```

Comment (blva) [reply internal/cli/atlas/search/nodes/list.go:59]: (context - since it's an atlas only new command we don't need mongocli)

**Review by blva — DISMISSED** [review]

File: internal/cli/atlas/search/search.go [hunk 3]
```diff
@@ -25,6 +26,7 @@ func Builder() *cobra.Command {
 		Short:   "Manage Atlas Search for your cluster.",
 	}
 	cmd.AddCommand(IndexesBuilder())
+	cmd.AddCommand(nodes.Builder())
```
Comment (gssbzn) [inline internal/cli/atlas/search/search.go:29]:
```suggestion
	cmd.AddCommand(
		IndexesBuilder(),
		nodes.Builder()
	)
```

**Review by blva — DISMISSED** [review]
🚀

**Review by gssbzn — APPROVED** [review]
LGTM don't forget to request review from docs

File: internal/cli/atlas/search/nodes/list.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 MongoDB Inc
```
Comment (tibulca) [inline internal/cli/atlas/search/nodes/list.go:1]:
```suggestion
// Copyright 2024 MongoDB Inc
```

File: internal/cli/atlas/search/nodes/list.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	return opts.Print(r)
+}
+
+// atlas clusters search nodes list [--projectId projectId] [--clusterName name].
+func ListBuilder() *cobra.Command {
+	opts := &ListOpts{}
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all Atlas Search nodes for a cluster.",
+		Long:  fmt.Sprintf(usage.RequiredRole, "Project Data Access Read/Write"),
```
Comment (tibulca) [inline internal/cli/atlas/search/nodes/list.go:65]: I think this API only needs `"Project Read Only"` access, or am I wrong?

Comment (jeroenvervaeke) [reply internal/cli/atlas/search/nodes/list.go]: I've update the command. Thanks

File: docs/atlascli/command/atlas-clusters-search-nodes.txt [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+   :widths: 20 10 10 60
+
+   * - Name
+     - Type
+     - Required
+     - Description
+   * - -h, --help
+     - 
+     - false
+     - help for nodes
```
Comment (lmkerbey-mdb) [inline docs/atlascli/command/atlas-clusters-search-nodes.txt:31]:
```suggestion
     - Help for nodes
```
Should this be capitalized?

Comment (jeroenvervaeke) [reply docs/atlascli/command/atlas-clusters-search-nodes.txt:31]: I don't think it should be. All other commands are lowercase

**Review by lmkerbey-mdb — APPROVED** [review]
LGTM.

We should look into fixing the lowercasing of the --help option descriptive text across all commands.

File: internal/cli/atlas/search/nodes/list_test.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mocks"
+	"github.com/mongodb/mongodb-atlas-cli/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-cli/internal/test"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115006/admin"
+)
+
+func TestList_Run(t *testing.T) {
```
Comment (fmenezes) [inline internal/cli/atlas/search/nodes/list_test.go:27]: we normally add tests for the builders as well

Comment (jeroenvervaeke) [reply internal/cli/atlas/search/nodes/list_test.go:27]: Missed your comment. Will address it in a separate PR

**Pull request merged**